app/websocket/app/main/model_manager.py

```diff
@@ -0,0 +1,188 @@
+import syft as sy
+
+import pickle
+import os
+
+from .persistence.models import db, SyftModel
+from sqlalchemy.exc import SQLAlchemyError, IntegrityError
+
+
+model_cache = dict()
+
+
+def cache_get_model(model_id: str):
+    """Checks the cache for a model. If model not found, returns None.
+    
+    Args:
+        model_id (str): Unique id representing the model. 
+
+    Returns:
+        A model encoded in ("ISO-8859-1"), else returns None.
+
+    """
+    model_ids = model_cache.keys()
+    if model_id in model_ids:
+        # model found
+        return model_cache[model_id]
+
+    return None
+
+
+def cache_save_model(serialized_model: bytes, model_id: str):
+    """Saves the model to cache. Will fail if a model with same id already exists.
+
+    Args:
+        serialized_model (bytes): The model object to be saved. encoded in ("ISO-8859-1").
+        model_id (str): The unique identifier associated with the model.
+
+    Returns:
+        True for success, False otherwise.
+
+    """
+    if cache_get_model(model_id) is None:
```

Comment:
it would be better to do: `if model_id not in model_cache:`. This way you don't need to actually get the object, just check if it is on the dict.